src/NovaVaultV2.sol

```diff
@@ -115,13 +115,13 @@ contract NovaVaultV2 is ReentrancyGuard, Ownable {
     ) internal {
         (bool success, ) = swapFacet.delegatecall(
             abi.encodeWithSignature(
-                "swapTokensGeneric(bytes32,string,string,address,uint256,(address,address,address,address,uint256,bytes,bool)[])",
+                "swapTokensSingleV3ERC20ToERC20(bytes32,string,string,address,uint256,(address,address,address,address,uint256,bytes,bool))",
                 "",
                 "integrator",
                 "referrer",
                 receiver,
                 minAmount,
-                swapData
+                swapData[0]
```

Comment:
could we also change the signature of our functions? no need to pass `LibSwap.SwapData[]` to `_swapTokensGeneric` if we just use the first element of the array

Reply:
otherwise, we could use the function [swapTokensMultipleV3ERC20ToERC20](https://github.com/lifinance/contracts/blob/18e47efcefcfab1b89a96da7c5706af11e5f108d/src/Facets/GenericSwapFacetV3.sol#L241) that takes an array of `LibSwap.SwapData[]` as argument

Reply:
Done.

I used the second option